ecl/hql/hqlgram2.cpp

```diff
@@ -12379,15 +12379,18 @@ IHqlExpression * PseudoPatternScope::lookupSymbol(IIdAtom * name, unsigned looku
 
 
 //---------------------------------------------------------------------------------------------------------------------
-
 extern HQL_API IHqlExpression * parseQuery(IHqlScope *scope, IFileContents * contents, HqlLookupContext & ctx, IXmlScope *xmlScope, IProperties * macroParams, bool loadImplicit, bool isRoot)
 {
+    static CriticalSection parseQueryCrit; // Temporary - see HPCC-21986
```

Comment:
If this wasn't temporary, I would suggest this moves outside the function - if inside a function the compiler needs to ensure it is only created once if called on multiple threads.  Gives me extra incentive to fix the issue properly.

Reply:
does the compiler actually intro a spinlock or mutex etc. to achieve that?